src/Bicep.Core/Registry/AzureContainerRegistryManager.cs

```diff
@@ -126,14 +120,14 @@ async Task<OciArtifactResult> DownloadManifestInternalAsync(bool anonymousAccess
                 }
              */
 
-            var manifest = new OciManifest(2, mediaType, artifactType, configDescriptor, layerDescriptors.ToImmutableArray(), annotations.Build());
-
-            using var manifestStream = new MemoryStream();
-            OciSerialization.Serialize(manifestStream, manifest);
-
-            manifestStream.Position = 0;
-            var manifestBinaryData = await BinaryData.FromStreamAsync(manifestStream);
-            var manifestUploadResult = await blobClient.SetManifestAsync(manifestBinaryData, artifactReference.Tag, mediaType: ManifestMediaType.OciImageManifest);
+            var manifest = new OciManifest(2, mediaType, artifactType, config, layerDescriptors.ToImmutableArray(), annotations.Build());
+            // var serializedManifest = JsonSerializer.Serialize(manifest,  OciManifestSerializationContext.Default.OciManifest);
+           
+            var manifestBinaryData = BinaryData.FromObjectAsJson(manifest, new JsonSerializerOptions{
+                PropertyNamingPolicy = JsonNamingPolicy.CamelCase,
+                Encoder = JavaScriptEncoder.UnsafeRelaxedJsonEscaping,
```

Comment:
Comment on why this encoder?

Reply:
Actually I was not planning on using this encoder at all, In fact this is an oversight since I should actually be using the options that are coming from the extension object. Good catch!